src/locator.rs

```diff
@@ -0,0 +1,143 @@
+use crate::errors::{Error, ErrorContext, Result, StdError};
+use std::borrow::Cow;
+use std::fmt;
+use std::sync::{Arc, LazyLock, Mutex};
+use std::time::{Duration, Instant};
+
+use async_trait::async_trait;
+use serde::Deserialize;
+use smart_default::SmartDefault;
+
+mod ip2location;
+mod ipapi;
+
+static LAST_AUTOLOCATE: Mutex<Option<AutolocateResult>> = Mutex::new(None);
+static LOCATOR: LazyLock<Mutex<Locator>> = LazyLock::new(|| Mutex::new(Locator::default()));
```

Comment:
instead of having a global locator, perhaps it could live in another existing construct like api?

Reply:
I'm okay with it being global

Reply:
On a second thought, storing `Locator` or `Arc<Locator>` in api may be slightly better, `set_global_locator` and the fact that it can be called from anywhere feels a bit weird.

Reply:
Moved it over to the `CommonApi`